### PR TITLE
Allow open registrations via config option and add a register link 

### LINKED ISSUE #4: Disable invite-only functionality
People should just be able to make accounts.

Extra credit if there's an admin-level option somewhere so we can turn invite-only back on without having to bounce the servers.

### PR DESCRIPTION
**Pull request by Zenchreal**
- add config option which controls whether registrations are open (if they're closed, the invite code is still required)
- add a register link to the header on the top right
- hide "Invite someone" when registrations are open

Fixes #4

### CODE REVIEW

**Comment by Zenchreal** [comment]
Thanks @rickhull

I ran the git push hook, which does code formatting/linting and runs the automated tests. Though it looks like there aren't any automated tests that cover the part of the code I touched.

Other than that, I did manual testing with the dev server. I only tested with Firefox, I could have tested Chrome too (but the changes are small enough that I don't think there's an issue).

**Comment by zorbathut** [comment]
(sorry it took a bit to get back to you, I didn't realize how the "review" feature worked. Got it sorted now, I think)

File: tildes/tildes/templates/login.jinja2 [hunk 1]
```diff
@@ -34,9 +34,9 @@
     </div>
   </form>
 
-  <hr>
+  <p class="text-small"><a href="/register">Register a new account.</a></p>
 
-  <p class="text-small">Tildes is currently in invite-only alpha, and you must be invited to be able to register. Most of the existing users have the ability to invite others, so if you know someone that has an account, you can ask them for an invite. Otherwise, please <a href="https://blog.tildes.net/announcing-tildes">read the announcement blog post</a> for more info about the site and instructions for requesting an invite from the site admin.</p>
+  <hr>
```
[the remark this answers is not in the document]
Comment (Zenchreal) [reply tildes/tildes/templates/login.jinja2:39]: Sounds good to me. I won't change the text of the old message just yet, as we can update that when we do #17

**Comment by zorbathut** [comment]
Looks good!

**Pull request merged**